runner/update_status.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+if [ "${RUNNER_CUSTOM_RBAC:-}" == "true" ]; then
+    export HTTPS_PROXY=
+    export https_proxy=
+
+    APISERVER=https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT_HTTPS}
+    SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount
+    NAMESPACE=$(cat ${SERVICEACCOUNT}/namespace)
+    TOKEN=$(cat ${SERVICEACCOUNT}/token)
+    CACERT=${SERVICEACCOUNT}/ca.crt
+
+    curl --silent -X PATCH --cacert ${CACERT} -H "Content-Type: application/merge-patch+json" -H "Authorization: Bearer ${TOKEN}" \
+        ${APISERVER}/apis/actions.summerwind.dev/v1alpha1/namespaces/${NAMESPACE}/runners/${HOSTNAME}/status \
+        -d "{\"status\":{\"phase\": \"$1\", \"message\": \"$2\"}}"
```

Comment:
We have to use `jq` to create the JSON that we are about to post, because JSON requires valid UTF-8 and certain bytes need to be escaped properly (just think of `"` in either `$1` or `$2`).

Local variables should always be written in lowercase. By convention the OS and shell use uppercase variables for environment and system variables, we do not want to interact with them.

Using long options and most of the time preferable over short options, as it makes the code more readable. Exceptions are very well known options (e.g. `rm -f`) or commands that do not provide long options (e.g. `jq -n`).

```suggestion
    serviceaccount=/var/run/secrets/kubernetes.io/serviceaccount
    namespace=$(cat $serviceaccount/namespace)
    token=$(cat $serviceaccount/token)
    jq -n --arg phase "$1" --arg message "${2:-}" '.status = $ARGS.named' | curl \
      --cacert $serviceaccount/ca.crt \
      --data @- \
      --header "Authorization: Bearer $token" \
      --header 'Content-Type: application/merge-patch+json' \
      --show-error \
      --silent \
      --request PATCH \
      "https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT_HTTPS/apis/actions.summerwind.dev/v1alpha1/namespaces/$namespace/runners/$HOSTNAME/status"
```